Mathlib/Data/Bitvec/Lemmas.lean

```diff
@@ -1,173 +1,115 @@
 /-
 Copyright (c) 2020 Simon Hudon. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Simon Hudon
+Authors: Simon Hudon, Harun Khan
 -/
-import Mathlib.Data.Bitvec.Defs
-import Mathlib.Data.Fin.Basic
-import Mathlib.Tactic.NormNum
 
-#align_import data.bitvec.basic from "leanprover-community/mathlib"@"008af8bb14b3ebef7e04ec3b0d63b947dee4d26a"
+import Mathlib.Data.Bitvec.Defs
 
 /-!
 # Basic Theorems About Bitvectors
 
-This file contains theorems about bitvectors and their coercions to
-`Nat` and `Fin`.
+This file contains theorems about bitvectors.
 -/
-namespace Bitvec
+
+namespace Std.BitVec
 
 open Nat
 
-theorem bitsToNat_toList {n : ℕ} (x : Bitvec n) : Bitvec.toNat x = bitsToNat (Vector.toList x) :=
-  rfl
-#align bitvec.bits_to_nat_to_list Bitvec.bitsToNat_toList
-
-attribute [local simp] Nat.add_comm Nat.add_assoc Nat.add_left_comm Nat.mul_comm Nat.mul_assoc
-
-attribute [local simp] Nat.zero_add Nat.add_zero Nat.one_mul Nat.mul_one Nat.zero_mul Nat.mul_zero
-
-local infixl:65 "++ₜ" => Vector.append
-
--- mul_left_comm
-theorem toNat_append {m : ℕ} (xs : Bitvec m) (b : Bool) :
-    Bitvec.toNat (xs++ₜb ::ᵥ Vector.nil) =
-      Bitvec.toNat xs * 2 + Bitvec.toNat (b ::ᵥ Vector.nil) := by
-  cases' xs with xs P
-  simp [bitsToNat_toList]; clear P
-  unfold bitsToNat
-  -- porting note: was `unfold List.foldl`, which now unfolds to an ugly match
-  rw [List.foldl, List.foldl]
-  -- generalize the accumulator of foldl
-  generalize h : 0 = x
-  conv in addLsb x b =>
-    rw [← h]
-  clear h
-  simp only [List.foldl_append, List.foldl_cons, List.foldl_nil]
-  induction' xs with x xs xs_ih generalizing x
-  · simp
-    unfold addLsb
-    simp [Nat.mul_succ]
-  · simp
-    apply xs_ih
-#align bitvec.to_nat_append Bitvec.toNat_append
```

Comment:
This align seems to have gone; I think it's a `noalign` since it's a very different statement to the new one, unless you want to implement `toNat_concat` (did `concat` land in Std?)

Reply:
`concat` did land in Std, but I don't think it's particularly useful to implement `toNat_concat`, let's just go with `noalign`